agent/src/beerocks/slave/backhaul_manager/backhaul_manager_thread.cpp

```diff
@@ -288,9 +298,17 @@ void backhaul_manager::platform_notify_error(int code, const std::string &error_
 
 void backhaul_manager::before_select()
 {
+    if (m_agent_ucc_listener) {
+        m_agent_ucc_listener->unlock();
```

Comment:
This deserves an explanation in the commit message and a comment which explains exactly what the `lock` and `unlock` are protecting.

Reply:
I explained it in the past, when I developed the ucc_listener.
It is needed since the ucc_listener runs in a different thread and shares the same memory with the backhaul manager.